python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,413 @@
+        Query the "Deep blank field catalogue : J/MNRAS/427/679" in Vizier.
+
+        Parameters
+        ----------
+        target : astropy.coordinates.SkyCoord
+            Sky coordinates near the field
+        radius : float
+            Search radius in degrees.
+
+        Reference
+        ---------
+        http://cdsarc.u-strasbg.fr/viz-bin/Cat?J/MNRAS/427/679
+        """
+        _table = Vizier.query_region(
+            catalog="J/MNRAS/427/679/blank_fld",
+            coordinates=target,
+            radius=radius * u.deg,
+        )
+
+        if len(_table) == 0:
+            self.log.info(
+                f"Could not find a field near {target} " f"within {radius} deg radius"
+            )
+            return None
+
+        _table = _table["J/MNRAS/427/679/blank_fld"]
+
+        coords = coordinates.SkyCoord(
+            ra=_table["RAJ2000"],
+            dec=_table["DEJ2000"],
+            unit=(u.hourangle, u.deg),
+            frame=coordinates.ICRS,
+        )
+
+        arg = target.separation(coords).argmin()
+
+        return coords[arg]
+
+    def confirm_sun_location(self):
+        """Confirm sun's elevation is safe for taking twilight flats."""
+        sun_coordinates = self.tcs.get_sun_azel()
+        where_sun = "setting" if (sun_coordinates[0] > 180) else "rising"
+        self.log.debug(
+            f" The azimuth of the {where_sun} Sun is {sun_coordinates[0]:.2f} deg \n"
+            f" The elevation of the Sun is {sun_coordinates[1]:.2f} deg"
+        )
+
+        if (where_sun < self.config.min_sun) or (where_sun > self.config.max_sun):
+            raise Exception(
+                f"Sun elevation {where_sun} is outside appropriate elvation limits. Aborting."
+            )
+
+    async def take_twilight_flats(self):
+
+        # Setup instrument filter
+        try:
+            await self.camera.setup_instrument(filter=self.get_instrument_filter())
+        except salobj.AckError:
+            self.log.warning(
+                f"Filter is already set to {self.get_instrument_filter()}. "
+                f"Continuing."
+            )
+
+        group_id = self.group_id if self.obs_id is None else self.obs_id
+
+        self.confirm_sun_location()
+
+        target = self.get_target_radec()
+
+        # get an empty field
+        search_area_degrees = 10
+
+        empty_field_coords = self.get_empty_field(target, radius=search_area_degrees)
+        self.log.info(
+            f"ICRS Empty field coordinates:\n"
+            f"  RA  = {empty_field_coords.ra.to_string(u.hour, sep=':')} ;"
+            f" DEC = {empty_field_coords.dec.to_string(u.degree, alwayssign=True, sep=':')}"
+        )
+
+        # slew to desired field
+        await self.tcs.slew_icrs(empty_field_coords.ra, empty_field_coords.dec)
+
+        # Take one 1s flat to calibrate the exposure time
+        self.log.info("Taking 1s flat to calibrate exposure time.")
+        exp_time = 1
+        await self.camera.take_flats(
```

Comment:
Maybe this should be a `take_acq` instead? We probably don't want to use this image for flat field right?